ci-operator/step-registry/baremetal/lab/pre/dns/baremetal-lab-pre-dns-commands.sh

```diff
@@ -1,2 +1,87 @@
-#!/bin/sh
-echo Hello world
+#!/bin/bash
+
+# This script modifies the following files in the auxiliary host:
+# - /opt/bind9_zones/zone
+# - /opt/bind9_zones/internal_zone.rev
+
+if [ -n "${LOCAL_TEST}" ]; then
+  # Setting LOCAL_TEST to any value will allow testing this script with default values against the ARM64 bastion @ RDU2
+  # shellcheck disable=SC2155
+  export NAMESPACE=test-ci-op AUX_HOST=openshift-qe-bastion.arm.eng.rdu2.redhat.com \
+      SHARED_DIR=${SHARED_DIR:-$(mktemp -d)} CLUSTER_PROFILE_DIR=~/.ssh INTERNAL_NET_IP=192.168.90.1
+fi
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+if [ -z "${AUX_HOST}" ]; then
+    echo "AUX_HOST is not filled. Failing."
+    exit 1
+fi
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-key")
+
+BASE_DOMAIN=$(<"${CLUSTER_PROFILE_DIR}/base_domain")
+
+# shellcheck disable=SC1090
+. <(yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")' < "${SHARED_DIR}"/external_vips.yaml)
+# shellcheck disable=SC2154
+if [ ${#api_vip} -eq 0 ] || [ ${#ingress_vip} -eq 0 ]; then
+  echo "Unable to parse VIPs"
+  exit 1
+fi
+
+DNS_FORWARD="#DO NOT EDIT; BEGIN $NAMESPACE
+api.${NAMESPACE} IN A ${api_vip}
+provisioner.${NAMESPACE} IN A ${INTERNAL_NET_IP}
+api-int.${NAMESPACE} IN A ${INTERNAL_NET_IP}
+*.apps.${NAMESPACE} IN A ${ingress_vip}"
+
+DNS_REVERSE_INTERNAL="#DO NOT EDIT; BEGIN $NAMESPACE"
+
+for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+  # shellcheck disable=SC1090
+  . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+  # shellcheck disable=SC2154
+  if [ ${#name} -eq 0 ] || [ ${#ip} -eq 0 ]; then
+    echo "Error when parsing the Bare Metal Host metadata"
+    exit 1
+  fi
+  DNS_FORWARD="${DNS_FORWARD}
+${name}.${NAMESPACE} IN A ${ip}"
+  DNS_REVERSE_INTERNAL="${DNS_REVERSE_INTERNAL}
+$(echo "${ip}." | ( while read -r -d . b; do rip="$b${rip+.}${rip}"; done; echo "$rip" )).in-addr.arpa. IN PTR ${name}.${NAMESPACE}.${BASE_DOMAIN}"
```

Comment:
For ipv6 it's `.ip6.arpa.`, but the v6 address has more forms such as
- `2001:0db8:85a3:0000:0000:8a2e:0370:7334`
- `2001:DB8::1`
- `2001:db8:0:0:0:0:2:1`

Reply:
Yes, I added a to do a few rows later. Either to do that as a follow up or before merging.

My only concern that could lead me to propose doing it in a follow-up is that I don't know how the ipv6 pool is managed in the upstream RH network and how we should do it for the self managed one, when available. It needs some thinking to get a stable solution IMO.

Reply:
Python3 standard library can handle that at ease.

```
>>> import ipaddress
>>> ipaddress.ip_address("2001:db8::1").reverse_pointer
'1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa'
>>> ipaddress.ip_address("192.168.11.1").reverse_pointer
'1.11.168.192.in-addr.arpa'
```

Reply:
I was wondering more about the ipv6 network topology configuration in the lab

Reply:
E.g., ipv6 could not reach external hosts in internet at the time I tried

Reply:
Let's sync offline for that once